packages/datastore-storage-adapter/__tests__/ExpoSQLiteDatabase.test.ts

```diff
@@ -0,0 +1,287 @@
+import ExpoSQLiteDatabase from '../src/ExpoSQLiteAdapter/ExpoSQLiteDatabase';
+import { openDatabase } from 'expo-sqlite';
+import { ParameterizedStatement } from '../src/common/types';
+import { HelperStack } from 'typedoc/dist/lib/output/utils/resources/helpers';
+
+// mock out the underlying implementation and only keep openDatabase().
+jest.mock('expo-sqlite', () => {
+	return {
+		__esModule: true,
+		openDatabase: jest.fn(() => mockDatabase),
+	};
+});
+
+// mock out the underlying implementation
+jest.mock('expo-file-system', () => 'File System');
+
+const mockDatabase = {
+	readTransaction: jest.fn(),
+	transaction: jest.fn(),
+};
+
+const mockTransaction = {
+	executeSql: jest.fn(),
+};
+
+const mockResultSet = jest.fn();
+
+const MOCK_SQL_ERROR = {
+	code: 400,
+	message: 'Error performing the transaction',
+};
+
+describe('ExpoSQLiteAdapter', () => {
+	const ExpoSQLiteAdapterInstance = new ExpoSQLiteDatabase();
+	ExpoSQLiteAdapterInstance.init();
+
+	beforeEach(() => {
+		mockDatabase.readTransaction.mockImplementation(func => {
+			func(mockTransaction);
+		});
+		mockDatabase.transaction.mockImplementation(func => {
+			func(mockTransaction);
+		});
```

Comment:
Don't think this is actually expected to change so you might be able to just move this up into the describe block instead of running it before each test.

Reply:
Yes, it can be moved up. Thanks